src/Core/UserRuleSettings/UserSettingsConstants.cs

```diff
@@ -0,0 +1,31 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+
+namespace SonarLint.VisualStudio.Core.UserRuleSettings;
+
+public static class UserSettingsConstants
```

Comment:
This should be a property of IUserSettingsProvider

Reply:
It can not be a property, because then I have to provide it in the UserSettingUpdater constructor to the SingleFileMonitor constructor creation. This makes the importing MEF constructor test to fail (because the property is null at the point when the constructor is called) So it has to be a static, so I thought a separate class makes more sense.

Reply:
Create export supports providing mocks with custom behavior. Ideally, the file watcher should be wrapped with a factory interface and mocked, so that we don't actually create the file watcher in tests.

Reply:
Will be done as part of the architecture refactoring, in https://sonarsource.atlassian.net/browse/SLVS-1301